### DISCUSSION TITLE
Why docker ?

### DISCUSSION THREAD

**Discussion by Schrebergarten03-links** [AzuraCast/AzuraCast#5673] [Q&A] [answered]
So im having constant troubles with the docker aspect of azuracast ...

3 times an normal update  wiped everything ...
backup doesnt work for me ...
all paths in the backend are totally different from actual path on host ... aka all in docker overlay2 obscoured by dif and merge etc

using linux for a few years on multiple servers im no noob .. but the complexity of docker is making it very hard to diagnose anything if there are errors ...

my hostserver  is running everything natively what azuracast needs (apache2 mariadb icecast liquidsoap)

So my nonhostile question:

Why are you using docker ? 
Wouldnt it be easier with just a directory in apache with php with a database and config files for icecast and liquidsoap ...
I dont understand the need for the added layer of docker ?

I dont want to start a war but i just wonder ... maybe since i´m no dev i cant see the whole picture.

Greetings and thanks for your hard work.

**Comment by BusterNeece** [comment, marked as answer]
@Schrebergarten03-links We've answered this several times in previous discussions, but to summarize: we're a very small team of developers, basically me working on the project heavily and another dev helping out when possible, and a community manager. We don't have the resources to support every possible combination of software on every host out there. Docker lets us ship a fully built image where all the software is in a known spot, won't conflict with other software on your system, and where we can easily guarantee that things will interact in a known way within that container.

If it's really important to you to avoid using Docker, you can use our Ansible installation method, but we don't provide the same level of support for it, as it's intended for "power users" who are familiar with the Linux shell and are able to diagnose and fix issues themselves if they arise.

Docker, with its images that our system automatically builds and deploys, is really the only way one developer can oversee several thousand installations running several tens of thousands of stations. Nobody who has complained about Docker has then gone on to improve our Ansible installation's support via pull requests, or has otherwise contributed any technical resources that would allow us to handle the greater support burden Ansible represents.

I would strongly recommend taking some time to learn about the Docker containerized filesystem. Once you're familiar with it, AzuraCast's setup makes perfect sense in that context. We are far from the only project moving to containers and solutions like Docker, so it could certainly be worthwhile to figure it all out.

**Comment by Schrebergarten03-links** [comment]
Okay well then im gonna find out about docker .. Since a lot of ppl use it cant be too bad.

Wow for 1.5 devs this project is fking awesome ;)

thy

**Comment by NotReeceHarris** [comment]
Why not implement a Kubernetes option as it is becoming a standardized technology
